### ISSUE TITLE
Request: Mappings to navigate to the first match in the next/prev file

### ISSUE THREAD

**Issue by samoshkin** [dyng/ctrlsf.vim#258]
I'm using "next" and "prev" mappings to navigate search matches one by one. It's like ":cnext" and ":cprev" commands for the Quickfix list. 

When search results list is long, it would be great to have mappings to navigate file by file, i.e. jump to first match in the prev/next file. Similar to what ":cnfile" and ":cpfile" commands do for the Quickfix list.

**Comment by kevinhwang91** [comment]
This pattern can do what you want.
```vim
function! CtrlSFAfterMainWindowInit()
    set wrap
    nnoremap <buffer><silent> [f :call <SID>nav_file(0)<CR>
    nnoremap <buffer><silent> ]f :call <SID>nav_file(1)<CR>
endfunction

function s:nav_file(forward) abort
    call search('^\%(\([0-9]\+\)\|\.\.\.\.\)\@!.\+\ze:$', a:forward ? 'W' : 'bW')
endfunction
```

**Comment by prabirshrestha** [comment]
+1 for this. It would be great if we could get a `<plug>` mapping for this.

**Comment by lyndhurst** [comment]
This feature along with the folding from #234 would be very nice additions to this plugin I think.

**Comment by dyng** [comment]
Hello all, I have implemented this feature in branch `feature/next-file-match`, please checkout and have a try. Any feedback is welcome.

My wife is so angry with me because I spent half of my weekend on coding and I need to calm her down now. 😅

**Comment by lyndhurst** [comment]
I am sorry, I cannot test extensively for a couple of weeks at least, but out of gratitude for your work (and respect for your wife's patience :) ), I installed it on my laptop to give it a try.

It installed without any problem, doing a few searches, I was able to navigate easily from file to file using `<C-N>` and `<C-P>` (the latter felt a bit slower to jump, but nothing significant). Files were folding, which is great for an overview of the results. I tried most of the features I normally use while doing a search without noticing anything wrong. On the other hand, I did not test any replace / edit action.

There is only one thing I noticed about folding, but I would have to check it out again later for the observation to be helpful to you. When I was trying to test some edge cases searching voluntarily for common words like 'if' or 'for' through a repo to get a lot of matches, sometimes, file folding would fail to enclose all the results and stop in between two matches from the same file. Even in those cases, `C-N` and `C-P` would still be working fine.

I know it is not very helpful because I could not see a pattern or anything obvious, and it might be due to my setup as well. I will try to check that particular issue again during the week, but I cannot commit to a particular timeframe, sorry. Let me know if there is something specific you would like me to test, and I will try to get back to you quickly.

Thanks again for the new feature.

**Comment by dyng** [comment]
Thank you for your feedback. I noticed same issue that folding failed occasionally, it is caused by half-done syntax recognition. Currently you can use `:syntax sync fromstart` to force vim parse syntax from start.

**Comment by lyndhurst** [comment]
Tried your suggestion, it works fine. I'll use the feature branch from next week on, on a more regular basis, and let you know after a while if anything comes up (or not). Thanks again.

**Comment by dyng** [comment]
Eventually I can't find a solution other than `:syntax sync fromstart`, it seems fine in this case.

**Comment by dyng** [comment]
I have used it for several days and found no problem, I think we can merge it and have a public testing.

**Comment by lyndhurst** [comment]
I found no problems either. And I can confirm that the "fix" for the folding issue is working fine for me.

**Comment by dyng** [comment]
Fixed in #318 .
